src/Microsoft.AspNet.StaticFiles/DirectoryBrowserMiddleware.cs

```diff
@@ -25,12 +26,21 @@ public class DirectoryBrowserMiddleware
         /// </summary>
         /// <param name="next">The next middleware in the pipeline.</param>
         /// <param name="options">The configuration for this middleware.</param>
-        public DirectoryBrowserMiddleware([NotNull] RequestDelegate next, [NotNull] IHostingEnvironment hostingEnv, [NotNull] DirectoryBrowserOptions options)
+        public DirectoryBrowserMiddleware([NotNull] RequestDelegate next,
+            [NotNull] IHostingEnvironment hostingEnv,
+            [NotNull] DirectoryBrowserOptions options,
+            [NotNull] IHtmlEncoder htmlEncoder)
```

Comment:
No, inside GenerateContextAsync it could do `context.ApplicationServices.GetRequiredService<IHtmlEncoder>()`